File: src/main/java/miragefairy2019/colormaker/MainColorMaker.java
package miragefairy2019.colormaker;

import java.io.IOException;

import miragefairy2019.colormaker.core.ColorIdentifier;
import miragefairy2019.colormaker.core.ImageLayer;
import miragefairy2019.colormaker.core.ImageLoader;
import miragefairy2019.mod.ModMirageFairy2019;
import mirrg.boron.util.struct.ImmutableArray;

public class MainColorMaker
{

	public static void main(String[] args) throws Exception
	{
		new WindowColorMaker().setVisible(true);
	}

	public static ImmutableArray<ImmutableArray<ImageLayer>> loadImages() throws IOException
	{
		ImageLoader imageLoader = new ImageLoader(ModMirageFairy2019.class.getClassLoader(), ModMirageFairy2019.MODID);
		return ImmutableArray.of(

			ImmutableArray.of(

				new ImageLayer(imageLoader.loadItemImage("fairy_layer0"), new ColorIdentifier("@skin")),
				new ImageLayer(imageLoader.loadItemImage("fairy_layer1"), new ColorIdentifier("#00BE00")),
				new ImageLayer(imageLoader.loadItemImage("fairy_layer2"), new ColorIdentifier("@darker")),
				new ImageLayer(imageLoader.loadItemImage("fairy_layer3"), new ColorIdentifier("@brighter")),
				new ImageLayer(imageLoader.loadItemImage("fairy_layer4"), new ColorIdentifier("@hair"))

			),
			ImmutableArray.of(

				new ImageLayer(imageLoader.loadItemImage("mirage_wisp_layer0"), new ColorIdentifier("@darker")),
				new ImageLayer(imageLoader.loadItemImage("mirage_wisp_layer1"), new ColorIdentifier("@skin")),
				new ImageLayer(imageLoader.loadItemImage("mirage_wisp_layer2"), new ColorIdentifier("@brighter")),
				new ImageLayer(imageLoader.loadItemImage("mirage_wisp_layer3"), new ColorIdentifier("@hair"))

			),
			ImmutableArray.of(

				new ImageLayer(imageLoader.loadItemImage("sphere_layer0"), new ColorIdentifier("@darker")),
				new ImageLayer(imageLoader.loadItemImage("sphere_layer1"), new ColorIdentifier("@hair")),
				new ImageLayer(imageLoader.loadItemImage("sphere_layer2"), new ColorIdentifier("@skin")),
				new ImageLayer(imageLoader.loadItemImage("sphere_layer3"), new ColorIdentifier("@brighter"))

			)

		);
	}

}

File: src/main/java/miragefairy2019/colormaker/MainColorMaker.kt
package miragefairy2019.colormaker

import miragefairy2019.mod.ModMirageFairy2019
import javax.imageio.ImageIO

private fun loadImage(name: String) = ImageIO.read(ModMirageFairy2019::class.java.classLoader.getResource("assets/${ModMirageFairy2019.MODID}/textures/items/$name.png")!!)

object MainColorMaker {
    @JvmStatic
    fun main(args: Array<String>) {
        WindowColorMaker(
            { loadImage(it) },
            listOf(
                layeredImageSettingOf(
                    LayerSetting("fairy_layer0", ColorExpression("@skin")),
                    LayerSetting("fairy_layer1", ColorExpression("#00BE00")),
                    LayerSetting("fairy_layer2", ColorExpression("@darker")),
                    LayerSetting("fairy_layer3", ColorExpression("@brighter")),
                    LayerSetting("fairy_layer4", ColorExpression("@hair"))
                ),
                layeredImageSettingOf(
                    LayerSetting("mirage_wisp_layer0", ColorExpression("@darker")),
                    LayerSetting("mirage_wisp_layer1", ColorExpression("@skin")),
                    LayerSetting("mirage_wisp_layer2", ColorExpression("@brighter")),
                    LayerSetting("mirage_wisp_layer3", ColorExpression("@hair"))
                ),
                layeredImageSettingOf(
                    LayerSetting("sphere_layer0", ColorExpression("@darker")),
                    LayerSetting("sphere_layer1", ColorExpression("@hair")),
                    LayerSetting("sphere_layer2", ColorExpression("@skin")),
                    LayerSetting("sphere_layer3", ColorExpression("@brighter"))
                )
            ),
            listOf("skin", "darker", "brighter", "hair")
        ).isVisible = true
    }
}

object MainColorMakerFairy {
    @JvmStatic
    fun main(args: Array<String>) {
        WindowColorMaker(
            { loadImage(it) },
            listOf(
                layeredImageSettingOf(
                    LayerSetting("fairy_layer0", ColorExpression("@skin")),
                    LayerSetting("fairy_layer1", ColorExpression("@dress")),
                    LayerSetting("fairy_layer2", ColorExpression("@dark")),
                    LayerSetting("fairy_layer3", ColorExpression("@bright")),
                    LayerSetting("fairy_layer4", ColorExpression("@hair"))
                )
            ),
            listOf("dress", "skin", "bright", "dark", "hair")
        ).isVisible = true
    }
}

object MainColorMakerSphere {
    @JvmStatic
    fun main(args: Array<String>) {
        WindowColorMaker(
            { loadImage(it) },
            listOf(
                layeredImageSettingOf(
                    LayerSetting("sphere_layer0", ColorExpression("@background")),
                    LayerSetting("sphere_layer1", ColorExpression("@plasma")),
                    LayerSetting("sphere_layer2", ColorExpression("@core")),
                    LayerSetting("sphere_layer3", ColorExpression("@highlight"))
                )
            ),
            listOf("core", "highlight", "background", "plasma")
        ).isVisible = true
    }
}
